fix: enable enrollment for courses with past upgrade deadline

arslanashraf7 opened the pull request:
#### Pre-Flight checklist

- [x] Screenshots and design review for any changes that affect layout or styling
  - [x] Desktop screenshots
- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested

#### What are the relevant tickets?
#4975 

#### What's this PR do?
- Enabled the enrollment for courses that have an upgrade deadline I the past
- fixes the display for `View on edX` and enrollment buttons

#### How should this be manually tested?
Steps from #4975 can be followed to test it

#### Screenshots (if appropriate) *(Updated)
**On dashboard( FA program -- When the user has not added/calculated a request for personal price and upgrade deadline for course is in past)**
<img width="1075" alt="Screenshot 2021-07-26 at 1 53 24 PM" src="https://user-images.githubusercontent.com/34372316/126962876-4a1fe2ff-ac27-45ea-80f8-a9fb83d15627.png">

**On dashboard( FA program -- When the user has added/calculated a request for personal price but its in pending state, and upgrade deadline for course is in past)**
<img width="1120" alt="Screenshot 2021-07-26 at 1 56 28 PM" src="https://user-images.githubusercontent.com/34372316/126962887-207f5fbb-c60c-4fe1-af6a-2fc3991e058d.png">
**On dashboard( FA program -- When the user personal price is approved and upgrade deadline for course is in past)**

<img width="1214" alt="Screenshot 2021-07-26 at 1 57 32 PM" src="https://user-images.githubusercontent.com/34372316/126962889-246cd38d-f061-403d-a126-32dadbf59640.png">


**On dashboard( Non-FA program)**
<img width="1073" alt="Screenshot 2021-07-26 at 1 58 57 PM" src="https://user-images.githubusercontent.com/34372316/126962892-104aed68-6fec-497f-a8b6-83341e1d97c7.png">

<img width="801" alt="Screenshot 2021-07-16 at 5 54 44 PM" src="https://user-images.githubusercontent.com/34372316/125951031-0e74501d-a86f-4c6e-ace8-c3298a75bb70.png">

**On program page**
<img width="519" alt="Screenshot 2021-07-16 at 5 48 23 PM" src="https://user-images.githubusercontent.com/34372316/125950866-e33e9988-a3d8-4572-aa93-028d4f527e78.png">

annagav reviewed (changes requested):
I think we should remove the option to pay in the dialog. Or disable it and add a message saying the the deadline to upgrade has passed, so they can't pay.

annagav commented:
Also, the are two messages are a somewhat confusing. The course progress message says "course stared" and the course status message says "course is starting"

arslanashraf7 commented:
> I think we should remove the option to pay in the dialog. Or disable it and add a message saying the the deadline to upgrade has passed, so they can't pay.

@annagav I've updated the PR with the below changes, let me know your thoughts:
1. Updated course status/progress messages in case of the future enrollment end date and past course start date.
2. Disabled the `Pay Now` button based on `Upgrade deadline`, if the `course upgrade deadline` is in past then the `Pay Now` button will be disabled for all the users either they are new or they come to upgrade the course due to other reason.

If the above perception is okay with you, I'll mark the PR ready for review.

annagav commented:
@pdpinch @briangrossman could you confirm that the copy changes are okay.

annagav commented:
@pdpinch @briangrossman Could you confirm the new dialog message "You can't pay for this course. The deadline to upgrade the course has passed."

pdpinch commented on static/js/components/CourseEnrollmentDialog.js:
```diff
@@ -121,6 +122,25 @@ export default class CourseEnrollmentDialog extends React.Component {
+    ) {
+      message = (
+        <p key="1">
+          You can't pay for this course, The deadline to upgrade the course has
+          passed.
```
```suggestion
          You cannot pay for this course at this time. The payment deadline has passed.
```

pdpinch commented:
Will this message apply to all programs, i.e. both the fa program (DEDP, with it's proctored exam system) and the non-fa programs (e.g. SCM) ?

Can you update the screenshot?

arslanashraf7 commented:
> Will this message apply to all programs, i.e. both the fa program (DEDP, with it's proctored exam system) and the non-fa programs (e.g. SCM) ?

Yes, it will be applied to both FA and Non-FA programs. However, the order of messages shown based on the conditions for FA programs can be seen in the screenshot descriptions.

Code reference for the order of messages for FA is listed below in orderly fashion from top-bottom:
- [Personal price is in pending state](https://github.com/mitodl/micromasters/blob/d4ffaa685127e22b0fae127e1b994d1f6dbb9d5c/static/js/components/CourseEnrollmentDialog.js#L89)
- [User has already paid](https://github.com/mitodl/micromasters/blob/d4ffaa685127e22b0fae127e1b994d1f6dbb9d5c/static/js/components/CourseEnrollmentDialog.js#L109)
- [Upgrade deadline has passed](https://github.com/mitodl/micromasters/blob/d4ffaa685127e22b0fae127e1b994d1f6dbb9d5c/static/js/components/CourseEnrollmentDialog.js#L125)
- [Personal price is approved/User had applied](https://github.com/mitodl/micromasters/blob/d4ffaa685127e22b0fae127e1b994d1f6dbb9d5c/static/js/components/CourseEnrollmentDialog.js#L144)
- [User has not calculated the personal price](https://github.com/mitodl/micromasters/blob/d4ffaa685127e22b0fae127e1b994d1f6dbb9d5c/static/js/components/CourseEnrollmentDialog.js#L165)
 
> Can you update the screenshot?

Updated

annagav reviewed (approved):
looks good 👍

annagav commented:
@umarmughal824 I am sorry for the misunderstanding, but Peter and Brian have to confirm your text changes.

briangrossman commented:
👍 
I'm fine with this as is.

I'm going to create a new issue with some minor text tweaks, but this PR can be released.
